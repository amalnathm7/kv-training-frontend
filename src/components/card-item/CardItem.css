.card-item {
    padding: 20px;
    display: flex;
    flex-direction: column;
    font-size: 16px;
    width: 16.5%;
    overflow: hidden;
}

.card-label {
    color: #8B8D9D;
}

.card-value {
    color: #434552;
    margin-top: 10px;
}