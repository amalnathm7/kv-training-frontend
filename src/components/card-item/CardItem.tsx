import React from 'react';
import './CardItem.css';
import StatusIcon from '../status-icon/StatusIcon';
import { StatusType } from '../../types/StatusType';
import { StatusColor } from '../../constants/statusColorConstants';

export type CardItemPropsType = {
  label: string;
  value: string;
  isStatus: boolean;
};

const CardItem: React.FC<CardItemPropsType> = (props) => {
  let status: StatusType;

  if (props.isStatus)
    status = {
      label: props.value,
      color: StatusColor[props.value.replace(/ /g, '_')]
    };

  return (
    <div className='card-item' data-testid='card-item-test'>
      <label className='card-label'>{props.label}</label>
      {!props.isStatus && <label className='card-value'>{props.value}</label>}
      {props.isStatus && (
        <div className='card-value'>
          <StatusIcon status={status}></StatusIcon>
        </div>
      )}
    </div>
  );
};

export default CardItem;
