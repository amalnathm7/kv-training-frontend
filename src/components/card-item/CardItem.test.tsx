import CardItem, { CardItemPropsType } from './CardItem';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Card Props Test', () => {
  test('If card rendered correctly', () => {
    const props: CardItemPropsType = {
      label: 'card-item',
      value: 'card-value',
      isStatus: true
    };

    const element = render(<CardItem {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If card status active rendered correctly', () => {
    const props: CardItemPropsType = {
      label: 'card-item',
      value: 'Active',
      isStatus: true
    };

    const element = render(<CardItem {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If card status inactive rendered correctly', () => {
    const props: CardItemPropsType = {
      label: 'card-item',
      value: 'Inactive',
      isStatus: true
    };

    const element = render(<CardItem {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If card status probation rendered correctly', () => {
    const props: CardItemPropsType = {
      label: 'card-item',
      value: 'Inactive',
      isStatus: true
    };

    const element = render(<CardItem {...props} />);

    expect(element).toMatchSnapshot();
  });
});
