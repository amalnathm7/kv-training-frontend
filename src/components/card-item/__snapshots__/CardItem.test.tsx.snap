// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Props Test If card file rendered correctly and file click works 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-item"
        data-testid="card-item-test"
      >
        <label
          class="card-label"
        >
          card-item
        </label>
        <div
          class="card-value"
          data-testid="card-file-test"
          style="cursor: pointer;"
        >
          <u>
            card-value
          </u>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-item"
      data-testid="card-item-test"
    >
      <label
        class="card-label"
      >
        card-item
      </label>
      <div
        class="card-value"
        data-testid="card-file-test"
        style="cursor: pointer;"
      >
        <u>
          card-value
        </u>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card Props Test If card rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-item"
        data-testid="card-item-test"
      >
        <label
          class="card-label"
        >
          card-item
        </label>
        <label
          class="card-value"
        >
          card-value
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-item"
      data-testid="card-item-test"
    >
      <label
        class="card-label"
      >
        card-item
      </label>
      <label
        class="card-value"
      >
        card-value
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card Props Test If card status rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-item"
        data-testid="card-item-test"
      >
        <label
          class="card-label"
        >
          card-item
        </label>
        <div
          class="card-value"
        >
          <div
            class="status-container"
            style="padding: 5px 20px 5px 20px; border-radius: 20px; width: 100px; display: flex; justify-content: center; font-size: 12px;"
          >
            <label>
              card-value
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-item"
      data-testid="card-item-test"
    >
      <label
        class="card-label"
      >
        card-item
      </label>
      <div
        class="card-value"
      >
        <div
          class="status-container"
          style="padding: 5px 20px 5px 20px; border-radius: 20px; width: 100px; display: flex; justify-content: center; font-size: 12px;"
        >
          <label>
            card-value
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
